From ddfd4b4ec50f9f5ce7ee10a91630b678eca89939 Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Thu, 25 Aug 2016 20:36:11 -0400
Subject: [PATCH 123/388] fsl_qbman: Only invalidate/prefech on PPC systems
 when PAMU is disabled

PAMU is only required for QBMan data stashing on PPC based systems.
Make sure we avoid the invalidate/prefetch on non PPC systems (i.e. ARM)
as no IOMMU assistance is needed for stashing operations.

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl_qbman/qman_low.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_low.h b/drivers/staging/fsl_qbman/qman_low.h
index 1fb8f98..8b1ac37 100644
--- a/drivers/staging/fsl_qbman/qman_low.h
+++ b/drivers/staging/fsl_qbman/qman_low.h
@@ -745,11 +745,12 @@ static inline void qm_dqrr_pvb_update(struct qm_portal *portal)
 	register struct qm_dqrr *dqrr = &portal->dqrr;
 	const struct qm_dqrr_entry *res = qm_cl(dqrr->ring, dqrr->pi);
 	DPA_ASSERT(dqrr->pmode == qm_dqrr_pvb);
-#ifndef CONFIG_FSL_PAMU
-        /*
-         * If PAMU is not available we need to invalidate the cache.
-         * When PAMU is available the cache is updated by stash
-         */
+#if (defined CONFIG_PPC || defined CONFIG_PPC64) && !defined CONFIG_FSL_PAMU
+	/*
+	 * On PowerPC platforms if PAMU is not available we need to
+	 * manually invalidate the cache. When PAMU is available the
+	 * cache is updated by stashing operations generated by QMan
+	 */
 	dcbi(res);
 	dcbt_ro(res);
 #endif
-- 
2.9.3

